arch/xtensa/src/esp32/Kconfig

```diff
@@ -130,6 +136,39 @@ config ESP32_ESP32DXWDXX
 	default n
 	select ESP32_DUAL_CPU
 
+choice ESP32_FLASH_MODE
```

Comment:
how do you think about putting "ESPTOOL" in the names of these things if they only affect esptool options?
with the current name it isn't clear to me if it's for tools or the nuttx binary itself.
probably not in this pr as it affects the existing names like ESP32_FLASH_16M though.